frappe/public/js/frappe/list/list_view.js

```diff
@@ -1354,6 +1354,12 @@ frappe.views.ListView = class ListView extends frappe.views.BaseList {
 	process_document_refreshes() {
 		if (!this.pending_document_refreshes.length) return;
 
+		const route = frappe.get_route() || [];
+		if (!cur_list || route[0] != "List" || cur_list.doctype != route[1]) {
+			// wait till user is back on list view before refreshing
+			return;
+		}
```

Comment:
This might break when/if user comes back but no new event is received, perhaps we can just trigger this function again when list view is rendered to process pending refreshes.

Reply:
it doesn't cause on every render it refreshes.